x/dex/keeper/product_lock.go

```diff
@@ -25,6 +25,9 @@ func (k Keeper) getProductLock(ctx sdk.Context, product string) *ordertypes.Prod
 
 // LockTokenPair locks token pair
 func (k Keeper) LockTokenPair(ctx sdk.Context, product string, lock *ordertypes.ProductLock) {
+	if lock == nil {
+		panic("a nil pointer appears")
```

Comment:
change(and subsequent ones as well)：
a nil pointer appears
to
the nil pointer is not expected